jsrc/array.hpp

```diff
@@ -52,6 +52,11 @@ pointer_to_values(array x) -> int64_t* {
     return reinterpret_cast<int64_t*>(reinterpret_cast<C*>(x) + x->kchain.k);
 }
 
+[[nodiscard]] inline auto
+is_sparse(array x) -> bool {
```

Comment:
Should be `noexcept`

Reply:
might as well add `constexpr`

Reply:
Actually, has `constexpr` any effect here? The `array`/`A` argument is a pointer, that shouldn't be known at compile time.

Reply:
still good practice to put on all functions